From a2dadfc0b1bac467d20e76935c6af6d9aeb19f6e Mon Sep 17 00:00:00 2001
From: Ricardo Salveti <ricardo@foundries.io>
Date: Tue, 21 Mar 2023 22:22:47 -0300
Subject: [PATCH 2/4] GENIO: arm: mediatek: mt8195: support CONFIG_SYSRESET

Don't define reset_cpu when CONFIG_SYSRESET is set by the user.

Signed-off-by: Ricardo Salveti <ricardo@foundries.io>
---
 arch/arm/mach-mediatek/mt8195/init.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/arch/arm/mach-mediatek/mt8195/init.c b/arch/arm/mach-mediatek/mt8195/init.c
index d6d058bbb7b..b2327896166 100644
--- a/arch/arm/mach-mediatek/mt8195/init.c
+++ b/arch/arm/mach-mediatek/mt8195/init.c
@@ -47,10 +47,12 @@ int mtk_soc_early_init(void)
 	return 0;
 }
 
+#ifndef CONFIG_SYSRESET
 void reset_cpu(ulong addr)
 {
 	psci_system_reset();
 }
+#endif
 
 int print_cpuinfo(void)
 {
-- 
2.39.1

